llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -611,6 +614,8 @@ class CallsiteContextGraph {
   /// Map from each context ID to the AllocationType assigned to that context.
   DenseMap<uint32_t, AllocationType> ContextIdToAllocationType;
 
+  std::map<uint32_t, uint64_t> ContextIdToTotalSize;
```

Comment:
Hmm, according to https://llvm.org/docs/ProgrammersManual.html#other-map-like-container-options:

"We never use containers like unordered_map because they are generally very expensive (each insertion requires a malloc)."

But I do see that I use unordered_map elsewhere in this file, probably due to a previous review comment. Is the advice in the documentation outdated?

In any case this should probably at least be a DenseMap, will make that change for now.

Reply:
Also documented the new map.